src/app/icd/ICDManager.cpp

```diff
@@ -15,12 +15,152 @@
  *    limitations under the License.
  */
 
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
 #include <app/icd/ICDManager.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/logging/CHIPLogging.h>
+#include <platform/ConnectivityManager.h>
+#include <platform/LockTracker.h>
+#include <platform/internal/CHIPDeviceLayerInternal.h>
 
 namespace chip {
 namespace app {
 
-ICDManager::ICDManager() {}
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::IcdManagement;
 
+void ICDManager::ICDManager::Init()
+{
+    uint32_t activeModeInterval;
+    IcdManagement::Attributes::ActiveModeInterval::Get(kRootEndpointId, &activeModeInterval);
+    VerifyOrDie(kFastPollingInterval.count() < activeModeInterval);
+    UpdateIcdMode();
+    UpdateOperationStates(ActiveMode);
+}
+
+bool ICDManager::SupportCheckInProtocol()
+{
+    bool success;
+    uint32_t featureMap;
+    success = (IcdManagement::Attributes::FeatureMap::Get(kRootEndpointId, &featureMap) == EMBER_ZCL_STATUS_SUCCESS);
+
+    return success ? ((featureMap & to_underlying(IcdManagement::Feature::kCheckInProtocolSupport)) != 0) : false;
+}
+
+void ICDManager::UpdateIcdMode()
+{
+    assertChipStackLockedByCurrentThread();
+
+    ICDMode tempMode = SIT;
+
+    // The Check In Protocol Feature is required and the slow polling interval shall also be greater than 15 seconds
+    // to run an ICD in LIT mode.
+    if (kSlowPollingInterval > kICDSitModePollingThreashold && SupportCheckInProtocol())
```

Comment:
The min polling interval is not a hard requirement. we don't need the check at this point.

Reply:
good point I thought it was a SHALL. Maybe it should be a SHALL. since it is not active code yet I won't touch it but let's decide if it should be a SHALL or stay a should. 

Either way, we will need a way to differentiate a LIT from an SIT at run time and both can have the CIP